mesonbuild/modules/gnome.py

```diff
@@ -1140,6 +1142,11 @@ def gtkdoc(self, state: 'ModuleState', args: T.Tuple[str], kwargs: 'GtkDoc') ->
                 raise InvalidArguments('gnome.gtkdoc: main_xml and main_xgml are exclusive arguments')
             main_file = main_xml
         moduleversion = kwargs['module_version']
+        if not kwargs['build_during_build']:
+            mlog.warning('"gnome.gtkdoc" documentation will be built during install instead of during build. '
+                         'This is discouraged and will be changed in a future meson version. '
+                         'To avoid building docs by default, wrap "gnome.gtkdoc" around a meson_option. '
+                         'If you are already doing this, this warning is safe to ignore')
```

Comment:
I think this is a place for `FeatureDeprecated.single_use()`, that way we give a deprecation warning, pluse then you only get the warning if you can actually fix it.

Reply:
> pluse then you only get the warning if you can actually fix it.

Sorry for my ignorance about meson, but how would that actually work? The way the new option is now implemented, its default will trigger the warning/deprecation message. In consequence, **all** current users of `gnome.gtkdoc` will get the warning. I feel this is totally suboptimal, but if the goal is to deprecate the feature in the long-run, I guess it is not nice to have maintainers which are doing things right, first add and then remove `build_during_build` flag. Does that make sense? Do you have any better suggestions on how to approach this?

Reply:
The Feature checks framework will consult the minimum version of meson specified in `project()`, if any, and if you currently declare a minimum version greater than the FeatureDeprecated, it will print a warning. Otherwise, it will remain silent on the theory that you currently support older versions of meson that don't support this new kwarg, so you cannot actually add it, and there is no use telling you that you should.

Reply:
Anyway, the current implementation of this warning actually warns you even if you do `build_during_build: false`. You'd need to have it default to None, check if `is None`, and then print the warning + update the kwargs to specify `False`.